lcm/lcm_trajectory.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <Eigen/Dense>
+#include <vector>
+#include <string>
+#include <unordered_map>
+#include "drake/systems/lcm/serializer.h"
+
+#include "dairlib/lcmt_saved_traj.hpp"
+
+namespace dairlib {
+
+class LcmTrajectory {
```

Comment:
Please add some documentation for the class. What does it store, how should it be used, etc.

One thing that surprised me was that the row-index relates to the datapoint, rather than the trajectory index.

Reply:
I would say in most software packages I've used, vector-valued time series data uses this format (rows ~ time indexing), but given that we usually think of state as a column vector I would say documenting is definitely in order.

Reply:
As for performance, it seems that the _rows_ are getting compact memory allocation, and memory operations are being done on a per-row basis. It would likely be more efficient to flip the rows/columns, but I don't know that we'll be handling data in quantities for which that will make much of a difference.

Reply:
Just adding my opinion here. I'm using to columns as the time index, and all of my code follows this rule. It will be great if we use columns for time.

Drake's cubic PiecewisePolynomial also follows the same rule 
https://drake.mit.edu/doxygen_cxx/classdrake_1_1trajectories_1_1_piecewise_polynomial.html#a81996df67cccc58f69bcff1f3e6b9671

Reply:
Changed datapoints so that the datatype is indexed by row, and times are indexed by column. Because I don't anticipate calling a save/load often per program or during a time-critical operation, I don't think it matters how we store the memory.